### PR TITLE
Adding Static code analysis in github actions

### PR DESCRIPTION
**Pull request by lekshmideepu**
Get the changed files and do the static code check.

### CODE REVIEW

**Review by heplesser — COMMENTED** [review]
Thanks, this looks good, just some tiny suggestions. Removal of .travis.yml would be in a separate PR, or do you want to include it here?

Have you tried this with some changes to C++/Python code or copyright headers that would trigger errors? Just to be sure that the checks really work.

**Comment by lekshmideepu** [comment]
@heplesser Thank you for reviewing this PR. I just removed the travis related files.
Tested the case with erroneous copyright headers for .py and .cpp files.Screenshot 2021-06-15 at 18 01 39

File: .github/workflows/nestbuildmatrix.yml [hunk 1]
```diff
@@ -98,6 +116,17 @@ jobs:
...
          run: sudo ldconfig
 
+       - name: Static Code Analysis
+         if: ${{ matrix.xNEST_BUILD_TYPE == 'STATIC_CODE_ANALYSIS' }}
+         run: |
+           chmod +x extras/gha_build.sh
+           ./extras/gha_build.sh
+         env:
+           xNEST_BUILD_TYPE: "STATIC_CODE_ANALYSIS"
+           #CHANGED_FILES: ${CHANGED_FILES}
```
[the remark this answers is not in the document]
Comment (lekshmideepu) [reply .github/workflows/nestbuildmatrix.yml]: Oh yes, I just removed that

File: .github/workflows/nestbuildmatrix.yml [hunk 2]
```diff
@@ -136,4 +165,4 @@ jobs:
            ./extras/gha_build.sh 2>&1 | tee gha_build.sh.log
            python extras/parse_build_log.py gha_build.sh.log ${{ github.workspace }}
          env:
-           xNEST_BUILD_TYPE: FULL
+           xNEST_BUILD_TYPE: FULL
```
[the remark this answers is not in the document]
Comment (lekshmideepu) [reply .github/workflows/nestbuildmatrix.yml]: done!

**Review by heplesser — APPROVED** [review]
Looks good!

**Comment by heplesser** [comment]
Labeled as Critical because we would lose static code checks without this PR when travis.org shuts down today.

**Comment by lekshmideepu** [comment]
Screenshot 2021-06-17 at 23 37 37

**Comment by heplesser** [comment]
Since Travis has now been turned off, I will merge this PR immediately. We need to monitor over the next few weeks that Github actions indeed report failing static checks on ill-formatted code.

**Comment by lekshmideepu** [comment]
> Since Travis has now been turned off, I will merge this PR immediately. We need to monitor over the next few weeks that Github actions indeed report failing static checks on ill-formatted code.

Sure!

**Pull request merged**